### DISCUSSION TITLE
Building tool: Vite vs SWC vs... Webpack?

### DISCUSSION THREAD

**Discussion by turolopezsanabria** [adevinta/spark#118] [Decision making]
## Decision 
We go with Vite and assess its feasibility and match for _Spark_ needs. 
If we arrive in March (official release of Storybook 7) and it's OK, we continue with it.
If not, we come back to this discussion.

## To take into consideration:

We will use Storybook 7, and they recommend going with Vite.
Read this one: https://storybook.js.org/blog/storybook-for-vite/

## Vite

**Pros**
- Uses ES Build and Rollup. Compatible with all plugins
- Faster for hot load
- Story
- zero config setup, installing StoryBook 7 
- In LBC uses rollup, faster to create the POC
- Time to market

**Cons / Risks**
- New tool, not as much support from the community as WebPack
- At this point of the project, the needs are very small, but this will probably grow as we progress over time.

## SWC

https://swc.rs/docs/usage/jest

**Pros**
- Speed

**Cons / Risks**
- Not ready for our need
- It couldn't manage ES modules correctly for now (they explain it in their doc)
- Isn’t mature enough
- It has random coverage issues (check [this PR](https://github.mpi-internal.com/leboncoin/frontend-web-rav-components/pull/5393))

**Comment by andresz1** [comment]
There was agreement regarding this. We are using `vite`
